pom.xml

```diff
@@ -3243,6 +3253,39 @@
           </dependency>
         </dependencies>
       </dependencyManagement>
+      <build>
+        <pluginManagement>
+          <plugins>
+            <plugin>
+              <groupId>net.alchim31.maven</groupId>
+              <artifactId>scala-maven-plugin</artifactId>
+              <configuration>
+                <args>
+                  <arg>-unchecked</arg>
```

Comment:
Can the additional args just be appended in this config?
Or do you need to override them to remove some, thus, need to duplicate some compiler flags here?
This is going to be OK, just making sure we're keeping it as simple as we can.

Reply:
OK~ Will  try it later. a little busy at the moment

Reply:
if we set `<args combine.children="append">` in Scala 2.13, the effective pom will be

```
<args combine.children="append">
                  <arg>-unchecked</arg>
                  <arg>-deprecation</arg>
                  <arg>-feature</arg>
                  <arg>-explaintypes</arg>
                  <arg>-target:jvm-1.8</arg>
                  <arg>-Xfatal-warnings</arg>
                  <arg>-Ywarn-unused:imports</arg>
                  <arg>-P:silencer:globalFilters=.*deprecated.*</arg>
                  <arg>-Wconf:cat=deprecation:wv,any:e</arg>
...
```

The are 3 flags shouldn't be in Scala 2.13

- `-P:silencer:globalFilters=.*deprecated.*`: This is the compilation option for `silencer-plugin`, compilation will fail because this plugin does not exist in Scala 2.13
- `-Xfatal-warnings` and `-Ywarn-unused:imports`: `unused-imports` check not enabled in Scala 2.13, compilation will fail  because scala.language.higherKinds no longer needs to be imported explicitly since Scala 2.13.1. Or need to add rules to suppress them.

So I think we may need to keep these duplicate flags here, or I haven't thought of an effective way yet